app/models/supplejack_api/search.rb

```diff
@@ -37,11 +37,22 @@ def self.schema_class
     # It will remove any invalid facets in order to avoid Solr errors
     #
     def facet_list
-      return @facet_list if @facet_list
+      facet_list = options[:facets].split(',').map { |f| f.strip.to_sym }
+      facet_list.keep_if { |f| self.class.model_class.valid_facets.include?(f) }
+
+      # This is to prevent users from requesting integer fields as facets
+      # because we do not have docValues built up for these fields facetting does not work.
+      # We do not have docValues because we are experiencing an issue with the facet counts being wrong
+      # between different Solr replicas.
```

Comment:
Nice comments ^^

Reply:
Thanks bro, definitely something strange that could be forgotten about.